### ISSUE TITLE
Missing try catch in osqp interface

### ISSUE THREAD

**Issue by mpowelson** [ros-industrial-consortium/trajopt_ros#205]
Opening an issue for this because the PR in which I thought I had fixed it did not work. See #204. I'm now getting `double free or corruption (out)`  immediately after the setup fails (and the exception gets caught). 
```
OSQPModel::~OSQPModel()
{
  // The osqp_workspace_ is managed by osqp but its members are not so must clean up.
  if (osqp_workspace_ != nullptr)
    osqp_cleanup(osqp_workspace_);  // double free or corruption (out)

}
```

**Comment by mpowelson** [comment]
@Levi-Armstrong On a scale of 1 to don't do that, how bad would doing this be? It is trading a crash for a memory leak in code that will hopefully be deprecated soon.

```
  auto ret = osqp_setup(&osqp_workspace_, &osqp_data_, &osqp_settings_);
  if (ret)
  {
    osqp_workspace_ = nullptr;
    throw std::runtime_error("Could not initialize OSQP: error " + std::to_string(ret));
  }
```

**Comment by mpowelson** [comment]
This was fixed in #204. See discussion over there, but I think we decided that there shouldn't be a memory leak.
